sqlglot/generator.py

```diff
@@ -347,6 +345,7 @@ class Generator:
     STRICT_STRING_CONCAT = False
     NORMALIZE_FUNCTIONS: bool | str = "upper"
     NULL_ORDERING = "nulls_are_small"
+    ESCAPE_LINE_BREAK = False
```

Comment:
Interesting. Didn't see this before 😄

Reply:
Yep - Toby added this a while ago because BQ doesn't allow actual newlines in single-quoted strings, i.e.

```sql
SELECT '
'
```

So if we ever encounter such a string, e.g. in Postgres, and want to transpile it to BQ we make sure to escape it:

```python
>>> import sqlglot
>>> sqlglot.transpile("""'
... '""", "postgres", "bigquery")
["'\\n'"]
```

Postgres:
```
georgesittas=# select '
georgesittas'# ';
 ?column?
----------
         +

(1 row)
```

BQ:
<img width="959" alt="Screenshot 2023-09-28 at 1 48 02 AM" src="https://github.com/tobymao/sqlglot/assets/46752250/e6a7665a-c465-4d6a-b20f-11bbe239eb31">

Reply:
This feels like it has the reproducibility problem if I then read the bigquery string in and pass `read="postgres"`, no?

Reply:
if you read it from a file, a true line break is \n, and what you see in the bq ui would actually be \\n

Reply:
What do you mean?

Reply:
Read in a bigquery select query with bigquery newline from a file, parse it as postgres. Is that the same result as passing in an ASCII 10 character, as you specified in your example?

Reply:
> Read in a bigquery select query with bigquery newline from a file, parse it as postgres.

I'm not sure if that premise makes sense? Why would you parse a BigQuery SQL file using Postgres? That's like feeding possibly invalid SQL into `parse_one` and expecting back a correct AST, right?

Reply:
Sorry, what I mean is: parse it as bigquery, transpile it to postgres. Are those equivalent when executed?

Reply:
Hmm I see what you mean, yea in that case the transpilation breaks because we generate `SELECt '\\n'` - not equivalent. I'll think about this one tomorrow, might be that the transpilation towards postgres is broken because it doesn't treat the backslash as an escape.